rotkehlchen/exchanges/binance.py

```diff
@@ -589,6 +605,121 @@ def _query_lending_balances(
 
         return balances
 
+    def query_lending_interests_history(
+            self,
+            cursor: 'DBCursor',
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> bool:
+        """Queries Binance lending interest history, transforms it into `HistoryBaseEntry` objects and save it in the database.
+
+        "DAILY" -> flexible savings.
+        "CUSTOMIZED_FIXED" -> fixed savings.
+        "ACTIVITY" -> fixed savings but as a result of token partners.
+
+        Lending Interest History Documentation:
+        https://binance-docs.github.io/apidocs/spot/en/#get-interest-history-user_data-2
```

Comment:
So I am quite pissed here. You are ignoring my questions/comments. I wrote here why was it written that the restriction is 30 days: https://github.com/rotki/rotki/pull/5681#discussion_r1126745716, when this restriction is not cheked anywhere.

You ignored it and just removed the comment and replaced it with a link to the docs. So get your shit together and answer the PR comments and **do not make me waste my time**.

1. Keep the link to the docs.
2. Re-add in the docstring the restriction that `end_ts` - `start_ts` need to be less than or equal to 30 days.
3. Explain to me **where is this restriction respected and enforced**? It's not anywhere and it should be in this function. This is right now directly called by the api, so it can have any value. So people wil query range [0 - now], get only last 30 days, save therange [0 - now] as queried in the DB and lose all older data.

Reply:
The restriction is enforced using `_api_query_list_within_time_delta()`. 

It uses a while to ensure that if the difference exceeds 30days e.g 34 days, it queries the first 30 days and subsequently queries the 4 days left. It is achieved using a while loop.